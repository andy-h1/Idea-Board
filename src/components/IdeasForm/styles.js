import styled from 'styled-components';

export const FormWrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
`;

export const Form = styled.form`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  text-transform: capitalize;
  background-color: #fdd835;
  border: 1px solid #90a4ae;
  border-radius: 0.5em;
  box-shadow: 0 3px 5px rgba(0, 0, 0, 0.18);
  margin: 1em;
  padding: 1em;
  width: 300px;
  height: 300px;
`;

export const Label = styled.label`
  margin: 1em;
`;

export const TitleInput = styled.input`
  margin: 0;
  padding: 1em;
  width: 250px;
  height: auto;
  font-family: 'Noto Sans JP', sans-serif;
`;

export const TextArea = styled.textarea`
  padding: 1em;
  width: 250px;
  height: auto;
  font-family: 'Noto Sans JP', sans-serif;
`;

export const CharacterCounter = styled.p`
  color: red;
  margin: 1em 0;
  font-size: 0.8em;
`;

export const ButtonWrapper = styled.div`
  display: flex;
  justify-content: flex-start;
  width: 100%;
`;

export const Button = styled.button`
  margin: 0 0.5em;
  padding: 0.5em;
  background-color: #2980b9;
  border-radius: 0.5em;
  border: none;
  color: white;
  box-shadow: 0 3px 5px rgba(0, 0, 0, 0.18);
  font-family: 'Open Sans', sans-serif;
  font-weight: 700;
  width: 100%;
  cursor: pointer;
`;

export const AddIdeaButton = styled.button`
  margin: 1em;
  padding: 0.5em;
  background-color: #2980b9;
  border-radius: 0.5em;
  border: none;
  color: white;
  box-shadow: 0 3px 5px rgba(0, 0, 0, 0.18);
  font-family: 'Open Sans', sans-serif;
  font-weight: 700;
  width: 250px;
  cursor: pointer;
`;
