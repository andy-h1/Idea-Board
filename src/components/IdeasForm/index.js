import React, { useContext, useState } from 'react';
import { IdeaContext } from '../../contexts/IdeaContext';
import * as S from './styles';

export const MAX_CHAR_LIMIT = 140;
export const MIN_CHAR_LIMIT = 120;

export const IdeasForm = () => {
  const { addIdea } = useContext(IdeaContext);
  const [title, setTitle] = useState('');
  const [description, setDescription] = useState('');
  const [characterCounter, setCharacterCounter] = useState(0);
  const [isAdding, setAdding] = useState(false);

  const handleTitle = (event) => {
    setTitle(event.target.value);
  };

  const handleDescription = (event) => {
    setDescription(event.target.value);
    setCharacterCounter(event.target.value.length);
  };

  const handleSubmit = (event) => {
    event.preventDefault();
    addIdea(title, description);
    setTitle('');
    setDescription('');
    setAdding(false);
  };

  return (
    <S.FormWrapper>
      {isAdding ? (
        <S.Form data-testid="ideasFormForm">
          <S.Label htmlFor="title">
            <S.TitleInput
              data-testid="ideasFormTitleInput"
              id="title"
              name="title"
              type="text"
              placeholder="Idea title"
              value={title}
              onChange={handleTitle}
              autoFocus
              autoComplete="off"
              required
            />
          </S.Label>

          <S.Label htmlFor="description">
            <S.DescriptionInput
              data-testid="ideasFormDescriptionInput"
              id="description"
              name="description"
              type="text"
              placeholder="Add your ideas here..."
              value={description}
              onChange={handleDescription}
              maxLength={MAX_CHAR_LIMIT}
              required
            />

            {characterCounter > MIN_CHAR_LIMIT && (
              <S.CharacterCounter>Characters remaining: {MAX_CHAR_LIMIT - characterCounter}</S.CharacterCounter>
            )}
          </S.Label>

          <S.ButtonWrapper>
            <S.Button
              data-testid="ideasFormSubmitButton"
              id="submit"
              name="add-idea"
              type="submit"
              value="add-idea"
              onClick={handleSubmit}
            >
              Add
            </S.Button>
            <S.Button data-testid="ideasFormCancelButton" type="button" name="cancel" onClick={() => setAdding(false)}>
              Cancel
            </S.Button>
          </S.ButtonWrapper>
        </S.Form>
      ) : (
        <S.AddIdeaButton data-testid="ideasFormAddIdeaButton" type="button" onClick={() => setAdding(true)}>
          Add Idea
        </S.AddIdeaButton>
      )}
    </S.FormWrapper>
  );
};
