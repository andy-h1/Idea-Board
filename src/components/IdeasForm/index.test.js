import React from 'react';
import { render } from '@testing-library/react';
import IdeasForm from '.';

describe('Ideas Form Component', () => {
  it('should render', () => {
    const { container } = render(<IdeasForm />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
