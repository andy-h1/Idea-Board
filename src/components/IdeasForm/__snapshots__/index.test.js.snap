// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IdeasForm should render 1`] = `
<div
  class="sc-bdnylx kFTTcc"
>
  <button
    class="sc-fujyUd TVRAI"
    data-testid="ideasFormAddIdeaButton"
    type="button"
  >
    Add Idea
  </button>
</div>
`;
